src/createQueryGroup.ts

```diff
@@ -171,19 +171,30 @@ export function createQueryGroup<
       .from(subscriptions.values())
       .map(subscription => subscription.retries)
     
-    retries.push(options?.retries)
+    retries.push(groupOptions?.retries)
 
     return reduceRetryOptions(retries)
   }
 
   function subscribe<
     TOptions extends QueryOptions<TAction>
   >(options?: TOptions): Query<TAction, TOptions> {
-    const removeSubscription = addSubscription(options)
+    const mergedOptions = {
+      ...groupOptions,
+      ...options,
+    }
+
+    const removeSubscription = addSubscription(mergedOptions)
 
     function dispose(): void {
+      const current = Array.from(subscriptions.values())
+      
       removeSubscription()
       setNextExecution()
+
+      for(const { onDispose } of current) {
+        onDispose?.()
+      }
```

Comment:
This doesn't seem right to me. When one subscription is disposed we're calling the `onDispose` of every subscription?

I expected this to only call the onDispose of the subscription being disposed of.